### PR TITLE
1865 Extract Re-IDed Napa DB to YAML, Validate and check referential integrity

### PR DESCRIPTION
**Pull request by mbthornton-lbl**
This PR provides a collection of MongoDB aggregation queries to search for referential integrity failures.

Test results from these queries are summarized in the parent ticket.

Based on these test results, this PR also closes:
- #1789 
- #1793
- #1835 
- #1836 
- #1814

### CODE REVIEW

File: assets/mongodb_queries/data_qc.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    """
+    Check referential integrity of slots in the MongoDB database using the queries in
+    in nmdb_schema/assets/mongodb_queries/data_qc.
+
+    Check for referential integrity of the following slots:
+        - part_of
+        - has_input
+        - has_output
+        - was_informed_by
+        - metagenome_annotation_id
```
Comment (aclum) [inline assets/mongodb_queries/data_qc.py:31]: i added info in the ticket on how to add checks for was_generated_by and has_calibration. Please add those to this python script.

File: assets/mongodb_queries/data_qc/applicable_class_was_generated_by_activity.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    var pipeline = [
+        {
+            $unwind: "$was_generated_by"
+        },
+        {
+            $lookup: {
+                from: "activity_set",
+                localField: "was_informed_by",
+                foreignField: "id",
+                as: "was_generated_by_docs"
```
Comment (aclum) [inline assets/mongodb_queries/data_qc/applicable_class_was_generated_by_activity.js:15]: This is confusing and mixes was_informed_by and was_generated_by.

Comment (aclum) [reply assets/mongodb_queries/data_qc/applicable_class_was_generated_by_activity.js:15]: local field should be 'was_generated_by'.
OmicsProcessing is also valid for a value for was_generated, I don't think that is accounted for with 'activity_set'

**Review by aclum — CHANGES_REQUESTED** [review]

File: assets/mongodb_queries/data_qc/applicable_class_was_generated_by_activity.js (hunk 2 above)
Comment (mbthornton-lbl) [reply assets/mongodb_queries/data_qc/applicable_class_was_generated_by_activity.js:15]: The documentation for `Class:DataObject` https://microbiomedata.github.io/nmdc-schema/DataObject/
has it as 0..1 with Range `Activity`

Comment (mbthornton-lbl) [reply assets/mongodb_queries/data_qc/applicable_class_was_generated_by_activity.js:15]: `was_generated_by` slot also shows Range: `Activity`

File: assets/mongodb_queries/data_qc.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env python3
```
[the remark this answers is not in the document]
Comment (mbthornton-lbl) [reply assets/mongodb_queries/data_qc.py]: I was having some difficulty getting this working so I have removed it from the PR and put it in a separate issue

File: assets/mongodb_queries/data_qc.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+# -*- coding: utf-8 -*-
```
[the remark this answers is not in the document]
Comment (mbthornton-lbl) [reply assets/mongodb_queries/data_qc.py]: Added these as req's to 
#1890

File: assets/mongodb_queries/data_qc.py (hunk 1 above)
Comment (mbthornton-lbl) [reply assets/mongodb_queries/data_qc.py]: Added queries to check was_generated_by and has_calibration relations.  I am pushing the scripting / automation of these queries to a separate issue

File: assets/mongodb_queries/data_qc/applicable_class_was_generated_by_activity.js (hunk 2 above)
Comment (mbthornton-lbl) [reply assets/mongodb_queries/data_qc/applicable_class_was_generated_by_activity.js]: Fixed above issue and added check for both Activity and OmicsProcessing

**Review by turbomam — APPROVED** [review]

**Comment by mbthornton-lbl** [comment]
This functionality in this PR also closes:
https://github.com/microbiomedata/nmdc_automation/issues/73

**Pull request merged**